homeassistant/components/rhasspy/stt.py

```diff
@@ -0,0 +1,197 @@
+"""
+Support for Rhasspy speech to text.
+
+For more details about this integration, please refer to the documentation at
+https://home-assistant.io/integrations/rhasspy/
+"""
+import io
+import logging
+from typing import List
+from urllib.parse import urljoin
+import wave
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.components import http
+from homeassistant.components.stt import Provider, SpeechMetadata, SpeechResult
+from homeassistant.components.stt.const import (
+    AudioBitRates,
+    AudioChannels,
+    AudioCodecs,
+    AudioFormats,
+    AudioSampleRates,
+    SpeechResultState,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN, SUPPORT_LANGUAGES
+
+# -----------------------------------------------------------------------------
+
+_LOGGER = logging.getLogger(__name__)
+
+# URL to POST WAV audio to
+CONF_SPEECH_URL = "speech_url"
+
+# Default settings
+DEFAULT_SPEECH_URL = "http://localhost:12101/api/speech-to-text"
+
+# Config
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({vol.Optional(CONF_SPEECH_URL): cv.url})
+
+# -----------------------------------------------------------------------------
+
+
+async def async_get_engine(hass, config, discovery_info):
+    """Set up Rhasspy speech to text component."""
+    provider = RhasspySTTProvider(hass, config)
+    _LOGGER.debug("Loaded Rhasspy stt provider")
+
+    # Register WAV speech to text test endpoint
+    hass.http.register_view(RhasspyWavView(hass))
+
+    return provider
+
+
+def get_speech_url(hass):
+    """Ties to get API URL from Rhasspy provider. Falls back to default."""
+    provider = hass.data.get(DOMAIN)
+    if provider is not None:
+        # Use provider URL
+        return urljoin(provider.api_url, "speech-to-text")
+
+    # Use default
+    return DEFAULT_SPEECH_URL
+
+
+# -----------------------------------------------------------------------------
+
+
+class RhasspySTTProvider(Provider):
+    """Rhasspy speech to text provider."""
+
+    def __init__(self, hass, conf):
+        """Create Rhasspy speech to text provider."""
+        self.hass = hass
+        self.config = conf
+
+        # URL to stream microphone audio
+        self.speech_url = conf.get(CONF_SPEECH_URL, None)
+        self.headers = {"Content-Type": "audio/wav"}
+
+    async def async_process_audio_stream(
+        self, metadata: SpeechMetadata, stream: aiohttp.StreamReader
+    ) -> SpeechResult:
+        """Process an audio stream to STT service.
+
+        Only streaming of content are allow!
+        """
+        if self.speech_url is None:
+            self.speech_url = get_speech_url(self.hass)
+
+        _LOGGER.debug("Receiving audio")
+        text_result = ""
+
+        try:
+            # First chunk is a WAV header (no frames)
+            header_chunk = True
+            with io.BytesIO() as wav_io:
```

Comment:
Like in function description, you need stream this directly to backend services. If you need an wave header, set the format to wave only. That is not our platform to deal with this things.